x-pack/plugin/enrich/src/test/java/org/elasticsearch/xpack/enrich/BasicEnrichTests.java

```diff
@@ -110,6 +110,62 @@ public void testIngestDataWithEnrichProcessor() {
+                    "-122.08193421363829 37.3879329075567," +
+                    "-122.08592534065245 37.3879329075567," +
+                    "-122.08592534065245 37.38501746624134))",
+                "zipcode", "94040"));
+            client().index(indexRequest).actionGet();
+            client().admin().indices().refresh(new RefreshRequest(SOURCE_INDEX_NAME)).actionGet();
+        }
+
+        String policyName = "my-policy";
+        EnrichPolicy enrichPolicy =
+            new EnrichPolicy(EnrichPolicy.GEO_MATCH_TYPE, null, List.of(SOURCE_INDEX_NAME), matchField, List.of(enrichField));
+        PutEnrichPolicyAction.Request request = new PutEnrichPolicyAction.Request(policyName, enrichPolicy);
+        client().execute(PutEnrichPolicyAction.INSTANCE, request).actionGet();
+        client().execute(ExecuteEnrichPolicyAction.INSTANCE, new ExecuteEnrichPolicyAction.Request(policyName)).actionGet();
+
+        String pipelineName = "my-pipeline";
+        String pipelineBody = "{\"processors\": [{\"enrich\": {\"policy_name\":\"" + policyName +
+            "\", \"field\": \"" + matchField + "\", \"target_field\": \"enriched\", \"max_matches\": 1 }}]}";
+        PutPipelineRequest putPipelineRequest = new PutPipelineRequest(pipelineName, new BytesArray(pipelineBody), XContentType.JSON);
+        client().admin().cluster().putPipeline(putPipelineRequest).actionGet();
+
+        BulkRequest bulkRequest = new BulkRequest("my-index");
+        IndexRequest indexRequest = new IndexRequest();
+        indexRequest.id("_id");
+        indexRequest.setPipeline(pipelineName);
+        indexRequest.source(Map.of(matchField, "37.386444, -122.083863")); // point within match boundary
+        bulkRequest.add(indexRequest);
+        BulkResponse bulkResponse = client().bulk(bulkRequest).actionGet();
+        assertThat("Expected no failure, but " + bulkResponse.buildFailureMessage(), bulkResponse.hasFailures(), is(false));
+        assertThat(bulkResponse.getItems().length, equalTo(1));
+        assertThat(bulkResponse.getItems()[0].getId(), equalTo("_id"));
+
+        GetResponse getResponse = client().get(new GetRequest("my-index", "_id")).actionGet();
+        Map<String, Object> source = getResponse.getSourceAsMap();
+        List<?> entries = (List<?>) source.get("enriched");
+        assertThat(entries, notNullValue());
+        assertThat(entries.size(), equalTo(1));
+
+        EnrichStatsAction.Response statsResponse =
+            client().execute(EnrichStatsAction.INSTANCE, new EnrichStatsAction.Request()).actionGet();
+        assertThat(statsResponse.getCoordinatorStats().size(), equalTo(1));
+        String localNodeId = getInstanceFromNode(ClusterService.class).localNode().getId();
+        assertThat(statsResponse.getCoordinatorStats().get(0).getNodeId(), equalTo(localNodeId));
+        assertThat(statsResponse.getCoordinatorStats().get(0).getRemoteRequestsTotal(), greaterThanOrEqualTo(1L));
```

Comment:
nit: this should be `equalTo(1)`, since only 1 index request is ingested.
This test suite also restarts the node between each test, so this should be ok.

Reply:
ah good catch. didn't update this.